filters/openpolicyagent/tracing.go

```diff
@@ -14,14 +20,43 @@ func init() {
 type tracingFactory struct{}
 
 type transport struct {
-	opa     *OpenPolicyAgentInstance
+	tracer     opentracing.Tracer
+	bundleName string
+	manager    *plugins.Manager
+
 	wrapped http.RoundTripper
 }
 
 func (*tracingFactory) NewTransport(tr http.RoundTripper, opts opatracing.Options) http.RoundTripper {
+	var tracer opentracing.Tracer
+
+	if len(opts) < 3 {
+		panic("insufficient tracing options provided for outbound http calls from Open Policy Agent")
```

Comment:
panic -> log.Fatal, but only fatal in case it's detected at startup. Everything that could happen at runtime -> Error instead.

Reply:
I do not like panic/fatal - we dont use any where exept command as this is not library friendly. Either we should change to return error or make it such that it works without options or ignores wrong options (and logs an error).

Reply:
We sadly cannot change the interface to return an error as this is coming from OPA itself. I also do not like panic/fatal but this way pretty much all OPA related tests will fail if the options are passed wrongly. 
I can also change this to just skip tracing if there is an error over here and log warnings but it will be tricky to have a meaningful test that actually validates this, the error will only happen in full integration testing with a filter and tracing.

Reply:
Maybe reach out to OPA folks and create a PR that returns an error

Reply:
I changed this so that it warns instead of failing which might degrade the level of tracing that is done. I also centralised the construction of the `Options` so that a mismatch would be covered with the tests for tracing.

The interface is public so would break other OPA users, filed an upstream issue here: https://github.com/open-policy-agent/opa/issues/6710 to see if we can change the interface...